pkg/reconciler/configmappropagation/configmappropagation.go

```diff
@@ -298,7 +298,20 @@ func (r *Reconciler) createOrUpdateConfigMaps(ctx context.Context, cmp *v1alpha1
 			//  OwnerReference will be removed when the knative.dev/config-propagation:copy label is not set in copy configmap
 			//  so that this copy configmap will not be deleted if cmp is deleted.
 			expected = current.DeepCopy()
-			expected.OwnerReferences = nil
+			// Delete the CMP owner reference, and keep all other owner references (if any)
+			if len(expected.OwnerReferences) > 0 {
+				index := -1
+				for i, owner := range expected.OwnerReferences {
+					if owner.UID == cmp.UID {
+						index = i
+						break
+					}
```

Comment:
I only find this way to delete an owner reference (by iterating the owner reference array). Not sure if there is other better way to do this.

Reply:
I would move this to a method in the controller `removeOwnerReference` so it's easier to read.

Rather than modifying the array in place, it might be simpler to copy each element to a new array, skipping the ref to remove. Code readability is more important than perf here IMO.

Reply:
That make sense, I just changed it. Thank you.